module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -211,17 +237,22 @@ public function getSearchesByChecksumAndOwner(
         UserEntityInterface|int|null $userOrId = null
     ): array {
         $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
-        $callback = function ($select) use ($checksum, $sessionId, $userId) {
-            $nest = $select->where
-                ->equalTo('checksum', $checksum)
-                ->and
-                ->nest
-                ->equalTo('session_id', $sessionId)->and->equalTo('saved', 0);
-            if (!empty($userId)) {
-                $nest->or->equalTo('user_id', $userId);
-            }
-        };
-        return iterator_to_array($this->getDbTable('search')->select($callback));
+        $dql = 'SELECT s FROM ' . $this->getEntityClass(SearchEntityInterface::class) . ' s '
+            . 'WHERE s.checksum = :checksum AND ';
+        $extraClauses = ['(s.sessionId = :sessionId AND s.saved = 0)'];
+        $params = [
+            'checksum' => $checksum,
+            'sessionId' => $sessionId,
+           ];
```

Comment:
I think you can simplify this back down to:

```suggestion
        $params = compact('checksum', 'sessionId');
```

...unless there's a reason you prefer not to.